docs/api.rst

```diff
@@ -13866,3 +13866,94 @@ Managing calendar reminders
   :statuscode 200: Data queried correctly.
   :statuscode 401: No user is currently logged in.
   :statuscode 500: Internal rotki error.
+
+Historical Balance Queries
+==============================
+
+.. http:get:: /api/(version)/balances/historical/(int:timestamp)
+
+  Gets a snapshot of asset balances at a specific timestamp.
+
+  Example Request:
+
+  .. http:example:: curl wget httpie python-requests
+
+    GET /api/(version)/balances/historical/1672531200 HTTP/1.1
+    Host: localhost:5042
+    Content-Type: application/json;charset=UTF-8
+
+  Example Response:
+
+  .. sourcecode:: http
+
+    HTTP/1.1 200 OK
+    Content-Type: application/json
+
+    {
+      "message": "",
+      "result": {
+        "BTC": {
+          "amount": "2.0",
+          "price": "20000"
+        },
+        "ETH": {
+          "amount": "10.0",
+          "price": "1500"
+        }
+      },
+      "status_code": 200
+    }
+
+  :resjson object result: Object mapping asset identifiers to their balance info
+  :resjson string amount: The amount of the asset held at the timestamp
+  :resjson string price: The price of the asset at the timestamp in the user's preferred currency
+  :statuscode 200: Historical balances returned
+  :statuscode 400: Invalid timestamp or malformed query
```

Comment:
Malformed query yes. But timestamp having no events, is not a bad request. It should be a 404. What we use for no data found.